Fix avatar upload to resize large profile pictures

HarshitR2004 opened the pull request:
# Description

This PR improves the user profile avatar upload functionality by ensuring that uploaded images are optimized before being saved. Currently, users can upload very large images, which may lead to unnecessary storage usage and slower page loads.

Fixes #12603 

## Changes include

- Automatically resize uploaded profile images if width or height exceeds 400px.
- Keep the original image if it is already ≤ 400px in both dimensions.
- Uses Pillow and the existing ImageTransform class to handle resizing.
- Added unit tests to verify that:
  - Large images are resized correctly.
  - Small images are not modified.
  - Images with width or height > 400px are resized proportionally.
  - Aspect ratio is preserved after resizing.

gasman commented on wagtail/admin/forms/account.py:
```diff
@@ -125,28 +129,55 @@ def __init__(self, *args, **kwargs):
+            return file 
+
+        transform = ImageTransform(size=(width, height))
+        new_transform = transform.resize((400, 400))
```
I don't think this is doing anything useful - if you check the code for `ImageTransform`, you'll see that we're just [writing `(400, 400)` to the `size` attribute](https://github.com/wagtail/wagtail/blob/f541cd5153cffcc484c162fb1d530cf350208336/wagtail/images/image_operations.py#L64) and immediately reading it back below. We also generally want to avoid importing things from `wagtail.images` within `wagtail.admin` - the images app can depend on admin, but not the other way round.

I also notice that this isn't preserving the aspect ratio of the original image, so if the user uploads a non-square image initially, it will end up distorted. For example, if a user uploads a 2000x1000 image, we'd want it to be resized to 800x400 (and _then_ maybe cropped to 400x400, although that's not essential as we can rely on the existing CSS for that). There's some existing logic for that elsewhere in `wagtail.images.image_operations` (such as in `MinMaxOperation`), but as mentioned above we should probably replicate that logic directly in this file rather than importing from the images app.

HarshitR2004 commented:
@gasman Thank you for your feedback, I have made the following changes

- Removed the dependency on wagtail.images
- Added aspect ratio-preserving resizing logic directly in clean_avatar(), similar to MinMaxOperation.
- Updated the tests accordingly

HarshitR2004 commented:
@gasman I have updated =tests to be filename-agnostic since avatar uploads now use InMemoryUploadedFile and Django’s storage backend may auto-generate unique filenames.

gasman commented on wagtail/admin/forms/account.py:
```diff
@@ -125,22 +129,59 @@ def __init__(self, *args, **kwargs):
             and self._original_avatar
             and (self._original_avatar != self.cleaned_data["avatar"])
         ):
-            # Call delete() on the storage backend directly, as calling self._original_avatar.delete()
```
Please don't remove comments without a good reason. (No worries, I'll reinstate these when I merge...)
